xmake/modules/package/tools/cmake.lua

```diff
@@ -419,15 +419,26 @@ function _get_configs_for_android(package, configs, opt)
     local ndk = get_config("ndk")
     if ndk and os.isdir(ndk) then
         local ndk_sdkver = get_config("ndk_sdkver")
-        local ndk_cxxstl = get_config("ndk_cxxstl")
         table.insert(configs, "-DCMAKE_TOOLCHAIN_FILE=" .. path.join(ndk, "build/cmake/android.toolchain.cmake"))
+        table.insert(configs, "-DANDROID_USE_LEGACY_TOOLCHAIN_FILE=OFF")
         table.insert(configs, "-DANDROID_ABI=" .. package:arch())
         if ndk_sdkver then
             table.insert(configs, "-DANDROID_PLATFORM=android-" .. ndk_sdkver)
             table.insert(configs, "-DANDROID_NATIVE_API_LEVEL=" .. ndk_sdkver)
         end
-        if ndk_cxxstl then
-            table.insert(configs, "-DANDROID_STL=" .. ndk_cxxstl)
+        -- https://cmake.org/cmake/help/latest/variable/CMAKE_ANDROID_STL_TYPE.html
+        local supported_runtimes = {
+            "none", "system", 
+            "gabi++_static", "gabi++_shared",
```

Comment:
package:has_runtime only support 

```
"gnustl_static", "gnustl_shared",
"c++_static", "c++_shared",
"stlport_static", "stlport_shared"
```

Reply:
I know but maybe it will be extended in the future, so I put all runtimes supported by this cmake option

Reply:
Please add them when xmake actually supports them. Also, the runtime names of cmake and ndk are not necessarily the same as the runtimes of xmake, and may need to be mapped, so don't add some unknown runtime names for now.

Reply:
Fair enough, done